subsys/bluetooth/mesh/gatt_cli.c

```diff
@@ -0,0 +1,349 @@
+		server->subscribe.notify = notify_func;
+		server->subscribe.value = BT_GATT_CCC_NOTIFY;
+		server->subscribe.ccc_handle = attr->handle;
+		server->subscribe.value_handle = attr->handle - 1;
+
+		err = bt_gatt_subscribe(conn, &server->subscribe);
+		if (err && err != -EALREADY) {
+			BT_DBG("Subscribe failed (err %d)", err);
+		} else {
+			BT_DBG("[SUBSCRIBED]");
+
+			server->gatt->link_open(conn);
+		}
+	}
+
+	return BT_GATT_ITER_STOP;
+}
+
+int bt_mesh_gatt_send(struct bt_conn *conn,
+		      const void *data, uint16_t len,
+		      bt_gatt_complete_func_t end, void *user_data)
+{
+	struct bt_mesh_gatt_server *server;
+
+	server = find_server(conn);
+
+	BT_DBG("%u bytes: %s", len, bt_hex(data, len));
+
+	return bt_gatt_write_without_response_cb(conn, server->data_in_handle,
+						 data, len, false, end, user_data);
+}
+
+static void gatt_connected(struct bt_conn *conn, uint8_t conn_err)
+{
+	struct bt_mesh_gatt_server *server;
+	struct bt_conn_info info;
+	int err;
+
+	bt_conn_get_info(conn, &info);
+	if (info.role != BT_CONN_ROLE_CENTRAL) {
+		return;
+	}
+
+	server = find_server(conn);
+
+	if (conn_err) {
+		BT_ERR("Failed to connect GATT Services(%u)", conn_err);
+
+		bt_conn_unref(conn);
+
+		(void)bt_mesh_scan_enable();
+
+		return;
+	}
+
+	BT_DBG("conn %p err 0x%02x", (void *)conn, conn_err);
+
+	server->gatt->connected(conn, server->user_data);
+
+	(void)bt_mesh_scan_enable();
+
+	server->discover.uuid = &server->gatt->srv_uuid.uuid;
+	server->discover.func = discover_func;
+	server->discover.start_handle = BT_ATT_FIRST_ATTRIBUTE_HANDLE;
+	server->discover.end_handle = BT_ATT_LAST_ATTRIBUTE_HANDLE;
+	server->discover.type = BT_GATT_DISCOVER_PRIMARY;
+	err = bt_gatt_discover(conn, &server->discover);
+	if (err) {
+		BT_ERR("Unable discover GATT Services (err %d)", err);
+	}
+}
+
+static void gatt_disconnected(struct bt_conn *conn, uint8_t reason)
+{
+	struct bt_conn_info info;
+	struct bt_mesh_gatt_server *server;
+
+	bt_conn_get_info(conn, &info);
+	if (info.role != BT_CONN_ROLE_CENTRAL) {
+		return;
+	}
+
+	server = find_server(conn);
+
+	server->gatt->disconnected(conn);
+
+	bt_conn_unref(conn);
+	(void)memset(server, 0, sizeof(struct bt_mesh_gatt_server));
+}
+
+int bt_mesh_gatt_connect(const bt_addr_le_t *addr,
+			 const struct bt_mesh_gatt_cli *gatt,
+			 void *user_data)
+{
+	int err;
+	struct bt_conn *conn;
+	struct bt_mesh_gatt_server *server;
+
+	/* Avoid interconnection between proxy client and server */
+	conn = bt_conn_lookup_addr_le(BT_ID_DEFAULT, addr);
+	if (conn) {
+		bt_conn_unref(conn);
+		return -EALREADY;
+	}
+
+	err = bt_mesh_scan_disable();
+	if (err) {
+		return err;
+	}
+
+	BT_DBG("Try to connect services");
+
+	err = bt_conn_le_create(addr, BT_CONN_LE_CREATE_CONN,
+				BT_LE_CONN_PARAM_DEFAULT, &conn);
+	if (err) {
+		BT_ERR("Connection failed (err:%d)", err);
+
+		(void)bt_mesh_scan_enable();
+
+		return err;
+	}
+
+	server = find_server(conn);
+	server->gatt = gatt;
+	server->user_data = user_data;
+
```

Comment:
Since you're not storing `conn` in the server context it seems like you should drop the reference given to you by `bt_conn_le_create()` here, i.e. add a `bt_conn_unref(conn);` call.

Reply:
Done.